BrainPortal/lib/boutiques_input_cache_cleaner.rb

```diff
@@ -36,44 +36,72 @@
 #       }
 #   }
 #
-# In the example above, any userfile cache selected for the file input
-# named 'my_input1' or 'my_input2' will be deleted after task execution unless CBRAIN
-# notices that another task uses same cache.
-# CBRAIN tries to handle conflict between tasks that share same file based on timestamps.
-# Yet this is a dangerous feature as it might not handle some of races, e.g. with legacy tools
+# In the example above, the cached content of any userfile(s) selected for the
+# file inputs named 'my_input1' or 'my_input2' will be deleted after the task
+# completes properly. This happens unless CBRAIN notices that another task
+# uses that input too. The module tries to handle conflict between tasks that
+# share same file based on the SyncStatus timestamps.
 module BoutiquesInputCacheCleaner
 
   def setup #:nodoc:
+
+    # BEFORE code: identify userfiles that are not synced at all
```

Comment:
`# BEFORE standard setup:  identify userfiles that are not synced at all `

would be clearer description. Also the class doc could mention that feature only work for nonsync files.

Reply:
I improved the documentation.